### PR TITLE
[bitnami/thanos] Adding sidecars to services with ingress

### PR DESCRIPTION
**Pull request by pburgisser**
**Description of the change**

Adding the possibility to add a sidecar container for each services that can be exposed with an ingress, so this services can be running behind a proxy like oAuth or a simple nginx proxy so access can be authenticated.

**Benefits**

Thanos components can be secured
**Possible drawbacks**

N/A

**Applicable issues**

N/A

**Additional information**

N/A

**Checklist** 

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/thanos/templates/bucketweb/deployment.yaml [hunk 1]
```diff
@@ -56,6 +56,9 @@ spec:
         fsGroup: {{ .Values.bucketweb.securityContext.fsGroup }}
       {{- end }}
       containers:
+        {{- if .Values.bucketweb.extraContainers }}
+{{ toYaml .Values.bucketweb.extraContainers | indent 8 }}
```
Comment (miguelaeh) [inline bitnami/thanos/templates/bucketweb/deployment.yaml:60]:
```suggestion
      {{ toYaml .Values.bucketweb.extraContainers | nindent 8 }}
```

Comment (miguelaeh) [reply bitnami/thanos/templates/bucketweb/deployment.yaml:60]: You can also use the tplvalue function from common

File: bitnami/thanos/templates/query-frontend/deployment.yaml [hunk 2]
```diff
@@ -61,6 +61,9 @@ spec:
         fsGroup: {{ .Values.queryFrontend.securityContext.fsGroup }}
       {{- end }}
       containers:
+        {{- if .Values.queryFrontend.extraContainers }}
+{{ toYaml .Values.queryFrontend.extraContainers | indent 8 }}
```
Comment (miguelaeh) [inline bitnami/thanos/templates/query-frontend/deployment.yaml:65]:
```suggestion
      {{ toYaml .Values.queryFrontend.extraContainers | nindent 8 }}
```

File: bitnami/thanos/templates/query/deployment.yaml [hunk 3]
```diff
@@ -61,6 +61,9 @@ spec:
         fsGroup: {{ $query.securityContext.fsGroup }}
       {{- end }}
       containers:
+        {{- if $query.extraContainers }}
+{{ toYaml $query.extraContainers | indent 8 }}
```
Comment (miguelaeh) [inline bitnami/thanos/templates/query/deployment.yaml:65]:
```suggestion
      {{ toYaml $query.extraContainers | nindent 8 }}
```

File: bitnami/thanos/templates/receive/statefulset.yaml [hunk 4]
```diff
@@ -60,6 +60,9 @@ spec:
         fsGroup: {{ .Values.receive.securityContext.fsGroup }}
       {{- end }}
       containers:
+        {{- if .Values.receive.extraContainers }}
+{{ toYaml .Values.receive.extraContainers | indent 8 }}
```
Comment (miguelaeh) [inline bitnami/thanos/templates/receive/statefulset.yaml:64]:
```suggestion
      {{ toYaml .Values.receive.extraContainers | nindent 8 }}
```

File: bitnami/thanos/templates/ruler/statefulset.yaml [hunk 5]
```diff
@@ -76,6 +76,9 @@ spec:
               mountPath: /data
       {{- end }}
       containers:
+        {{- if .Values.ruler.extraContainers }}
+{{ toYaml .Values.ruler.extraContainers | indent 8 }}
```
Comment (miguelaeh) [inline bitnami/thanos/templates/ruler/statefulset.yaml:80]:
```suggestion
      {{ toYaml .Values.ruler.extraContainers | nindent 8 }}
```

File: bitnami/thanos/values.yaml [hunk 6]
```diff
@@ -125,6 +125,27 @@ query:
   ##
   # existingSDConfigmap:
 
+  ## Optional additional containers (sidecar)
```
Comment (miguelaeh) [inline bitnami/thanos/values.yaml:128]:
```suggestion
  ## Optional additional containers (sidecar)
  ##
```

File: bitnami/thanos/values.yaml [hunk 7]
```diff
@@ -125,6 +125,27 @@ query:
...
+  extraContainers: []
```
Comment (miguelaeh) [inline bitnami/thanos/values.yaml:129]:
```suggestion
  extraContainers: []
  ## Example:
```

File: bitnami/thanos/values.yaml [hunk 8]
```diff
@@ -537,6 +559,27 @@ queryFrontend:
   ##
   hostAliases: []
 
+  ## Optional additional containers (sidecar)
```
Comment (miguelaeh) [inline bitnami/thanos/values.yaml:562]:
```suggestion
  ## Optional additional containers (sidecar)
  ##
```

File: bitnami/thanos/values.yaml [hunk 9]
```diff
@@ -537,6 +559,27 @@ queryFrontend:
...
+  extraContainers: []
```
Comment (miguelaeh) [inline bitnami/thanos/values.yaml:563]:
```suggestion
  extraContainers: []
  ## Example:
```

File: bitnami/thanos/values.yaml [hunk 10]
```diff
@@ -861,6 +905,27 @@ bucketweb:
   ##
   timeout: 5m
 
+  ## Optional additional containers (sidecar)
```
Comment (miguelaeh) [inline bitnami/thanos/values.yaml:908]:
```suggestion
  ## Optional additional containers (sidecar)
  ##
```

File: bitnami/thanos/values.yaml [hunk 11]
```diff
@@ -861,6 +905,27 @@ bucketweb:
...
+  extraContainers: []
```
Comment (miguelaeh) [inline bitnami/thanos/values.yaml:909]:
```suggestion
  extraContainers: []
  ## Example:
```

File: bitnami/thanos/values.yaml [hunk 12]
```diff
@@ -1717,6 +1783,27 @@ ruler:
   ##
   # clusterName:
 
+  ## Optional additional containers (sidecar)
```
Comment (miguelaeh) [inline bitnami/thanos/values.yaml:1786]:
```suggestion
  ## Optional additional containers (sidecar)
  ##
```

File: bitnami/thanos/values.yaml [hunk 13]
```diff
@@ -1717,6 +1783,27 @@ ruler:
...
+  extraContainers: []
```
Comment (miguelaeh) [inline bitnami/thanos/values.yaml:1787]:
```suggestion
  extraContainers: []
  ## Example:
```

File: bitnami/thanos/values.yaml [hunk 14]
```diff
@@ -2071,6 +2159,27 @@ receive:
   #   - endpoints:
   #     - "127.0.0.1:10901"
 
+  ## Optional additional containers (sidecar)
```
Comment (miguelaeh) [inline bitnami/thanos/values.yaml:2162]:
```suggestion
  ## Optional additional containers (sidecar)
  ##
```

File: bitnami/thanos/values.yaml [hunk 15]
```diff
@@ -2071,6 +2159,27 @@ receive:
...
+  extraContainers: []
```
Comment (miguelaeh) [inline bitnami/thanos/values.yaml:2163]:
```suggestion
  extraContainers: []
  ## Example:
```

**Review by miguelaeh — CHANGES_REQUESTED** [review]
Hi @pburgisser ,
Thank you very much for the PR!
Please check the suggestions I added

**Comment by pburgisser** [comment]
Hi @miguelaeh,

Thanks for the suggestions, they all have been integrated into the code!

Cheers

**Review by miguelaeh — CHANGES_REQUESTED** [review]
Hi @pburgisser ,
The `toYaml` lines are still not properly indented, could you please put them at the same indent level as the `if` line?
Also, it would be nice if you can use the `common.tplvalues` function instead of `toYaml`

**Comment by pburgisser** [comment]
Hi @miguelaeh,
No problem!
Thanks

**Review by miguelaeh — APPROVED** [review]
Thank you very much for applying the changes!
LGTM!

**Comment by miguelaeh** [comment]
Hi @pburgisser ,
It seems the linter complains about trailing spaces in the lines you added to the values.yaml, could you take a look? https://github.com/bitnami/charts/pull/6259/checks?check_run_id=2498630267

**Comment by pburgisser** [comment]
Hi @miguelaeh,

Yep, I've just fix them 

Thanks!

**Pull request merged**